tests/integ/modin/index/test_timedelta_index_methods.py

```diff
@@ -128,3 +128,29 @@ def test_timedelta_total_seconds():
     native_index = native_pd.TimedeltaIndex(TIMEDELTA_INDEX_DATA)
     snow_index = pd.Index(native_index)
     eval_snowpark_pandas_result(snow_index, native_index, lambda x: x.total_seconds())
+
+
+@pytest.mark.parametrize("skipna", [True, False])
+@pytest.mark.parametrize("data", [[1, 2, 3], [1, 2, 3, None], [None], []])
+@sql_count_checker(query_count=1)
+def test_timedelta_index_mean(skipna, data):
+    native_index = native_pd.TimedeltaIndex(data)
+    snow_index = pd.Index(native_index)
+    native_result = native_index.mean(skipna=skipna)
+    snow_result = snow_index.mean(skipna=skipna)
+    # Special check for NaN because Nan != Nan.
+    if pd.isna(native_result):
+        assert pd.isna(snow_result)
+    else:
+        assert snow_result == native_result
+
+
+@sql_count_checker(query_count=0)
+def test_timedelta_index_mean_invalid_axis():
+    native_index = native_pd.TimedeltaIndex([1, 2, 3])
+    snow_index = pd.Index(native_index)
+    with pytest.raises(IndexError, match="tuple index out of range"):
+        native_index.mean(axis=1)
+    # Snowpark pandas raises ValueError instead of IndexError.
+    with pytest.raises(ValueError, match="axis should be 0 for TimedeltaIndex.mean"):
+        snow_index.mean(axis=1)
```

Comment:
Unsure if the `to_pandas()` is necessary here with SQL Simplifier turned off.
```suggestion
        snow_index.mean(axis=1).to_pandas()
```